### PR TITLE
Add occupancy branches + standalone improvements (rebase PR45, PR36, and PR55)

### PR DESCRIPTION
**Pull request by ariostas**
This is a rebase of PRs #45 #36 and #55. I grouped them since they are standalone-only and relatively simple.

For #45, I simply adapted to the new SoA stuff.

For #36, I made the changes in that PR, but with additional changes that I'll comment in the code.

For #55, I copied the current readme and updated things. For this one, we could alternatively do it at the very end in case there are more changes.

### CODE REVIEW

File: RecoTracker/LSTCore/src/LSTESData.cc
Comment (ariostas) [inline RecoTracker/LSTCore/src/LSTESData.cc:16]: I changed the initialization here since the IB was complaining about a `nullptr`.

File: RecoTracker/LSTCore/standalone/setup.sh
Comment (ariostas) [inline RecoTracker/LSTCore/standalone/setup.sh:11]: I changed it to check if a version was set in case we want to override it in the CI. Also, now it uses the command Slava suggested instead of a hard-coded path for the release in case we want to use an IB version.

File: RecoTracker/LSTCore/standalone/README.md
Comment (slava77) [inline RecoTracker/LSTCore/standalone/README.md:224]: is this used in the CI?
I recall that we've updated to something more fresh than 21034

File: RecoTracker/LSTCore/standalone/setup.sh
Comment (ariostas) [reply RecoTracker/LSTCore/standalone/setup.sh:11]: Actually, now I'm thinking that this might lead to confusion. It should check for something more explicit like `FORCED_CMSSW_VERSION`

File: RecoTracker/LSTCore/standalone/README.md
Comment (ariostas) [reply RecoTracker/LSTCore/standalone/README.md:224]: Good catch! It's 29834.1

File: RecoTracker/LSTCore/standalone/README.md
Comment (ariostas) [reply RecoTracker/LSTCore/standalone/README.md:24]: Yeah, that would be good. Maybe it can just be `master_LST`, we submit PRs to CMSSW from there, and every time they get merged we sync it with the upstream master.

File: RecoTracker/LSTCore/standalone/setup.sh
Comment (ariostas) [reply RecoTracker/LSTCore/standalone/setup.sh]: Fixed

File: RecoTracker/LSTCore/standalone/README.md
Comment (VourMa) [reply RecoTracker/LSTCore/standalone/README.md:24]: I think mkFit has an established way of doing that that works nicely. @slava77 could probably give us the details.

**Comment by GNiendorf** [comment]
I haven't checked to see if this runs properly, but it looks fine otherwise (for my portion of the PR).

File: RecoTracker/LSTCore/standalone/README.md
Comment (slava77) [reply RecoTracker/LSTCore/standalone/README.md:24]: mkFit is using `master` in trackreco repository directly for the internal PRs. It's resynced whenever necessary. Also, in mkFit we never merge PRs directly: the same topic branch is submitted to the upstream and once it's reviewed and merged, it will appear as merged in trackreco as well (except that not all developers do it and we end up with PRs that look unmerged but actually are).
there is no CI in mkFit and using `master` is more straightforward.
So, I'm not sure that in our case the is a good reference in mkFit practices.

With CI things get complicated and we need to discuss/decide what's a better mode.

Smth like `CMSSW_14_2_0_pre4_LST_X` that's explicitly coupled to a pre-release is more clear for developers and the CI. The downside is this requires explicit updates to a new branch and pre-release. 

Using `master` for a target can be OK, but the CI would need to be updated to figure out the latest working release and do a rebase of the topic branch for a test, which is problematic. 

Another question is do we want to merge in SegmentLinking/cmssw/master-or-whatever before the full review in the upstream?

Comment (ariostas) [reply RecoTracker/LSTCore/standalone/README.md:24]: I was under the assumption that we would submit to CMSSW in batches so that things can move quicker. Either way, I can figure out a way to get the CI to pick the right release if needed.

Comment (slava77) [reply RecoTracker/LSTCore/standalone/README.md:24]: > I was under the assumption that we would submit to CMSSW in batches so that things can move quicker.

It's practical for our todos from #75 
After that part settles it may be more practical to avoid batching.

File: RecoTracker/LSTCore/standalone/README.md
Comment (VourMa) [inline RecoTracker/LSTCore/standalone/README.md:31]: Now that the LST PR is in CMSSW, these steps can in principle be skipped. Maybe add a comment that this is only needed to get some bleeding-edge development from some branch?

File: RecoTracker/LSTCore/standalone/README.md
Comment (VourMa) [inline RecoTracker/LSTCore/standalone/README.md:52]: Maybe update this to at least be consistent with the one mention in the section just above?

File: RecoTracker/LSTCore/standalone/README.md
Comment (VourMa) [inline RecoTracker/LSTCore/standalone/README.md:158]: Do we even need this section? Isn't that already covered above ("Setting up LST" section)?

...I know, probably a question to myself back in the day, but maybe we can improve now...

**Review by VourMa — COMMENTED** [review]
Thanks for the rebase, @ariostas! The PR looks good, I have a couple of minor comments. I guess the most significant one would be to settle on the main branch, but this could even go in before that, if we decide on that.

File: RecoTracker/LSTCore/standalone/README.md
Comment (ariostas) [reply RecoTracker/LSTCore/standalone/README.md]: Thank you, Manos! I addressed all these comments.

**Review by VourMa — APPROVED** [review]

**Comment by ariostas** [comment]
> we talked about getting something better named

What if we do `CMSSW_14_2_X_LST_batch1_devel`?

**Comment by ariostas** [comment]
> What if we do `CMSSW_14_2_X_LST_batch1_devel`?

I just saw that Gavin merged another PR, so let's stick with the current name for now. We could just rename it before submitting the CMSSW PR if necessary.

**Comment by slava77** [comment]
> > What if we do `CMSSW_14_2_X_LST_batch1_devel`?
> 
> Maybe just drop `batch1`?

in my comment I had in mind that this branch will become a topic branch for a PR to CMSSW.

There's a benefit of also having a branch for general LST integration; there I think both `batch1` and `devel` can be dropped.

**Comment by VourMa** [comment]
> in my comment I had in mind that this branch will become a topic branch for a PR to CMSSW.
> There's a benefit of also having a branch for general LST integration; there I think both batch1 and devel can be dropped.

Sorry, I must have missed your comment because I hadn't reloaded the page previously. I think that this makes sense in general but I would like to understand some details with a specific example:

Should that "general LST integration" branch be the target for my "HLT config" PR?
If so, how can the the "rebased developments" PR go in separate PRs in CMSSW but also be reviewed in parallel with the "HLT config" PR?
If not, which branch should be my target?

**Comment by slava77** [comment]
> Should that "general LST integration" branch be the target for my "HLT config" PR?
> If so, how can the the "rebased developments" PR go in separate PRs in CMSSW but also be reviewed in parallel with the "HLT config" PR?
> If not, which branch should be my target?

in #94
the only change in LST is in RecoTracker/LSTCore/interface/Constants.h, while the rest is elsewhere. It may be better to not combine LST-only changes with quite extensive updates outside. So, for a CMSSW review target, I think it's better to submit #94 separately

**Comment by ariostas** [comment]
The last batch of rebases have a conflict with this PR. Is this ready to be merged? If not, I'll wait until this is merged.

**Pull request merged**
